travis: don't launch the windows build for pull requests

Currently we launch the windows VM, and then abort it for a subset of build requests. It would be clearer (and more efficient) to not launch the windows build at all for those cases.

I think this can be done via `matrix.include` and conditionals:
https://docs.travis-ci.com/user/multi-os/
https://docs.travis-ci.com/user/conditional-builds-stages-jobs/

There are two issues now:

1. The Windows build fails due to some new code in Contributebot:

```
# gocloud.dev/internal/contributebot [gocloud.dev/internal/contributebot.test]
.\sync.go:177:25: undefined: unix.SIGTERM
FAIL	gocloud.dev/internal/contributebot [build failed]
```

It's concerning that it was so easy to miss this. I just happened to notice it. Are Windows builds really that much flakier than others? Maybe we should reconsider allowing failures....

2. The Windows builds on the master branch never seem to actually run tests. They do the checkout, install `wire`, and then hang until they fail. (I got the error above by making a pull request enabling Windows builds for PRs).

See any build [here](https://travis-ci.com/google/go-cloud/builds).

--- a/.travis.yml
+++ b/.travis.yml
@@ -56,11 +56,7 @@ jobs:
         verbose: true  # temporarily, while verifying
         on:
           branch: master
-  # Windows is in early access and has been unreliable. Don't block builds on it,
-  # but provide data (for now).
-  allow_failures:
-    - os: windows
-      
+
 stages:
   - test
   - name: imports

--- a/internal/contributebot/go.mod
+++ b/internal/contributebot/go.mod
@@ -3,7 +3,7 @@ module gocloud.dev/internal/contributebot
 replace gocloud.dev => ../..
 
 require (
-	cloud.google.com/go v0.30.0
+	cloud.google.com/go v0.34.0
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible
 	github.com/google/go-cmp v0.2.0
 	github.com/google/go-github v17.0.0+incompatible

--- a/internal/contributebot/sync.go
+++ b/internal/contributebot/sync.go
@@ -156,7 +156,8 @@ func gitSync(ctx context.Context, gitPath string, dst, src remoteGitBranch) erro
 // runCommand starts a subprocess and waits for it to finish. If the subprocess
 // exits with failure, the returned error's message will include the combined
 // stdout and stderr. If the context is cancelled, then runCommand sends the
-// subprocess SIGTERM (this differs from CommandContext, which sends SIGKILL).
+// subprocess os.Interrupt (this differs from CommandContext, which sends
+// os.Kill).
 func runCommand(ctx context.Context, dir string, exe string, args []string) error {
 	c := exec.Command(exe, args...)
 	c.Dir = dir

--- a/internal/contributebot/sync.go
+++ b/internal/contributebot/sync.go
@@ -27,7 +27,6 @@ import (
 	"strings"
 
 	"github.com/google/go-github/github"
-	"golang.org/x/sys/unix"
 )
 
 // syncParams is the set of parameters to syncPullRequest.
@@ -174,7 +173,7 @@ func runCommand(ctx context.Context, dir string, exe string, args []string) erro
 		select {
 		case <-ctx.Done():
 			// Ignore error: if we can't signal, it's likely because the process already exited.
-			_ = c.Process.Signal(unix.SIGTERM)
+			_ = c.Process.Signal(os.Interrupt)
 		case <-waitDone:
 		}
 		// No further synchronization back to the main goroutine, because signal is uninterruptable and

